packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -0,0 +1,327 @@
+        ? expandedIds.filter((id) => id !== nodeId)
+        : [...expandedIds, nodeId],
+    );
+  };
+
+  const collapsedPageIds = useMemo(
+    () =>
+      treeItems
+        .filter((item) =>
+          isPageCollapsed(item, treeItems, expandedPageIds, activeId),
+        )
+        .map(({ page }) => page.entityId),
+    [treeItems, expandedPageIds, activeId],
+  );
+
+  const pagesFlatIdList = useMemo(
+    () => treeItems.map(({ page }) => page.entityId),
+    [treeItems],
+  );
+
+  useEffect(() => {
+    // We only update the state when everything has been loaded.
+    // If the request fails, pages will be reordered to their original state
+    if (!loading) {
+      setTreeItems(getTreeItemList(data));
+      setInitialLoading(false);
+    }
+  }, [data, loading]);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 5,
+      },
+    }),
+  );
+
+  const dragDepth = useMemo(
+    () => Math.round(offsetLeft / IDENTATION_WIDTH),
+    [offsetLeft],
+  );
+
+  const projected =
+    activeId && overId
+      ? getProjection(treeItems, collapsedPageIds, activeId, overId, dragDepth)
+      : null;
+
+  useEffect(() => {
+    document.body.style.setProperty("cursor", activeId ? "grabbing" : "");
+  }, [activeId]);
+
+  const resetState = () => {
+    setOverId(null);
+    setActiveId(null);
+    setOffsetLeft(0);
+  };
+
+  const handleDragStart = ({
+    active: { id: activeItemId },
+  }: DragStartEvent) => {
+    setActiveId(activeItemId);
+    setOverId(activeItemId);
+  };
+
+  const handleDragMove = ({ delta }: DragMoveEvent) => {
+    setOffsetLeft(delta.x);
+  };
+
+  const handleDragOver = ({ over }: DragOverEvent) => {
+    setOverId(over?.id ?? null);
+  };
+
+  const handleDragEnd = async ({ active, over }: DragEndEvent) => {
+    resetState();
+
+    if (projected && over) {
+      const { depth, parentPageEntityId } = projected;
+
+      const clonedItems = [...treeItems];
+
+      const overIndex = clonedItems.findIndex(
+        ({ page }) => page.entityId === over.id,
+      );
+      const activeIndex = clonedItems.findIndex(
+        ({ page }) => page.entityId === active.id,
+      );
+      const activeTreeItem = clonedItems[activeIndex];
+
+      if (
+        activeTreeItem &&
+        (activeTreeItem.depth !== depth || active.id !== over.id)
+      ) {
+        clonedItems[activeIndex] = {
+          page: {
+            ...activeTreeItem.page,
+            parentPageEntityId,
+          },
+          depth,
+        };
+
+        const sortedItems = arrayMove(clonedItems, activeIndex, overIndex);
+
+        const newIndex = sortedItems
+          .filter(({ page }) => page.parentPageEntityId === parentPageEntityId)
+          .findIndex(({ page }) => page.entityId === activeId);
+
+        const parentSortedItems = sortedItems.filter(
+          ({ page }) => page.parentPageEntityId === parentPageEntityId,
+        );
+
```

Comment:
Correct me if I'm wrong, but you're doing this twice in a row?

```ts
sortedItems.filter(({ page }) => page.parentPageEntityId === parentPageEntityId)
```

Reply:
Nice spot, thanks. 3b4b6ccaf15e3b400f934ec8605d0a32998adeee